docs/aws/audit/kmsmonitoring/rules/secret_rotation_interval_remediation.mdx

```diff
@@ -1,6 +1,235 @@
+
+   ```sh
+   aws secretsmanager create-secret \
+       --name MySecretName \
+       --secret-string file://mysecret.json \
+       --kms-key-id alias/my-kms-key \
+       --rotation-lambda-arn arn:aws:lambda:region:account-id:function:MyRotationLambda \
+       --rotation-rules AutomaticallyAfterDays=30
+   ```
+
+2. **Enable Rotation for an Existing Secret:**
+   If you have an existing secret, you can enable rotation using the `rotate-secret` command.
+
+   ```sh
+   aws secretsmanager rotate-secret \
+       --secret-id MySecretName \
+       --rotation-lambda-arn arn:aws:lambda:region:account-id:function:MyRotationLambda \
+       --rotation-rules AutomaticallyAfterDays=30
+   ```
+
+3. **Update Rotation Rules for an Existing Secret:**
+   To update the rotation rules for an existing secret, use the `update-secret` command with the `--rotation-rules` parameter.
+
+   ```sh
+   aws secretsmanager update-secret \
+       --secret-id MySecretName \
+       --rotation-rules AutomaticallyAfterDays=30
+   ```
+
+4. **Verify Rotation Configuration:**
+   To ensure that the rotation configuration is correctly set, you can describe the secret and check its rotation rules using the `describe-secret` command.
+
+   ```sh
+   aws secretsmanager describe-secret \
+       --secret-id MySecretName
+   ```
+
+   Look for the `RotationEnabled` and `RotationRules` fields in the output to verify that rotation is enabled and configured correctly.
+
+By following these steps, you can ensure that your secrets in AWS Secrets Manager are rotated frequently, thereby enhancing the security of your secrets.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of not rotating Secrets Manager secrets frequently in AWS KMS using Python scripts, you can follow these steps:
+
+### 1. **Set Up AWS SDK for Python (Boto3)**
+First, ensure you have the AWS SDK for Python (Boto3) installed. You can install it using pip if you haven't already:
+
+```bash
+pip install boto3
+```
+
+### 2. **Create a Python Script to Enable Automatic Rotation**
+You can create a Python script to enable automatic rotation for your secrets in AWS Secrets Manager. Below is an example script:
+
+```python
+import boto3
+from botocore.exceptions import ClientError
+
+# Initialize a session using Amazon Secrets Manager
+client = boto3.client('secretsmanager')
+
+def enable_secret_rotation(secret_id, rotation_lambda_arn):
+    try:
+        response = client.rotate_secret(
+            SecretId=secret_id,
+            RotationLambdaARN=rotation_lambda_arn,
+            RotationRules={
+                'AutomaticallyAfterDays': 30  # Rotate every 30 days
+            }
+        )
+        print(f"Rotation enabled for secret: {secret_id}")
+    except ClientError as e:
+        print(f"Error enabling rotation for secret {secret_id}: {e}")
+
+# Replace 'your-secret-id' and 'your-lambda-arn' with your actual secret ID and Lambda ARN
+secret_id = 'your-secret-id'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>